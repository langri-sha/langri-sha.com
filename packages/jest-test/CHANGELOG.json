{
  "name": "@langri-sha/jest-test",
  "entries": [
    {
      "date": "Wed, 24 Apr 2024 22:03:36 GMT",
      "version": "0.2.0",
      "tag": "@langri-sha/jest-test_v0.2.0",
      "comments": {
        "patch": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/jest-test",
            "commit": "8d67a370e43a304d96b4e84a0eb773a9f7d4fecd",
            "comment": "feat(workspace): Use PNPM local references"
          }
        ],
        "minor": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/jest-test",
            "commit": "f9dcfcb1ce5b663131c076f5364425c03f757710",
            "comment": "feat(workspace): Update package metadata"
          }
        ]
      }
    }
  ]
}
