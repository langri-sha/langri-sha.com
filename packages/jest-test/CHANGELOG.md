# Change Log - @langri-sha/jest-test

This log was last generated on Fri, 03 May 2024 12:54:50 GMT and should not be manually modified.

<!-- Start content -->

## 0.7.1

Fri, 03 May 2024 12:54:50 GMT

### Patches

- fix(workspace): Correct published paths (filip.dupanovic@gmail.com)

## 0.7.0

Fri, 03 May 2024 10:44:32 GMT

### Minor changes

- feat(jest-test): Configure package build (filip.dupanovic@gmail.com)

### Patches

- fix(workspace): Sync packages (filip.dupanovic@gmail.com)
- fix(workspace): Revise build command (filip.dupanovic@gmail.com)
- fix(workspace): Sync packages (filip.dupanovic@gmail.com)

## 0.2.0

Wed, 24 Apr 2024 22:03:36 GMT

### Minor changes

- feat(workspace): Update package metadata (filip.dupanovic@gmail.com)

### Patches

- feat(workspace): Use PNPM local references (filip.dupanovic@gmail.com)
