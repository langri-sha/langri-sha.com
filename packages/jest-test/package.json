{
  "name": "@langri-sha/jest-test",
  "version": "0.1.0",
  "license": "MIT",
  "main": "src/index.ts",
  "dependencies": {
    "@jest/globals": "29.7.0",
    "tempy": "1.0.1"
  },
  "devDependencies": {
    "@langri-sha/tsconfig": "0.1.0"
  },
  "peerDependencies": {
    "jest": "^29.0.0",
    "typescript": "^5.0.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
