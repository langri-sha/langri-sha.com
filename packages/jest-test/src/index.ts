export * from '@jest/globals'
export { default as tempy } from 'tempy'
export { default as nock } from 'nock'
