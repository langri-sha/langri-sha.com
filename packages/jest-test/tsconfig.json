{
  "$schema": "https://json.schemastore.org/tsconfig",
  "extends": "@langri-sha/tsconfig/project.json",
  "references": [{ "path": "../tsconfig" }]
}
