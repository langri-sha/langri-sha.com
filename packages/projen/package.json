{
  "name": "@langri-sha/projen",
  "version": "0.1.0",
  "license": "MIT",
  "main": "src/index.ts",
  "devDependencies": {
    "@langri-sha/tsconfig": "workspace:*",
    "projen": "0.81.1"
  },
  "peerDependencies": {
    "projen": "^0.81.1",
    "typescript": "^5.0.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
