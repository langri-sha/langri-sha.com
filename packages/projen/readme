# @langri-sha/projen

Collection of [projen] templates for bootstraping monorepos and workspace projects.

[projen]: https://projen.dev/
