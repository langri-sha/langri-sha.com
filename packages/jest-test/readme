# @langri-sha/jest-test

Provides some useful helpers that are commonly used for authoring tests.

## Usage

Install the required dependencies:

```sh
npm install -D jest @langri-sha/jest-test
```

Then import your Jest test dependencies from here:

```js
// some.test.js
import { expect, test, tempy } from '@langri-sha/jest-test'

test(/*...*/)
```

## See

- [`@jest/globals`]
- [`tempy`]

[`@jest/globals`]: https://jestjs.io/docs/api
[`tempy`]: https://github.com/sindresorhus/tempy
