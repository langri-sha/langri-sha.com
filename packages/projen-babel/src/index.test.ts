import { synthSnapshot } from 'projen/lib/util/synth'
import { expect, test } from '@langri-sha/jest-test'

import { Project } from 'projen'
import { Babel } from './index'

test('defaults', () => {
  const project = new Project({
    name: 'test-project',
  })

  new Babel(project)

  expect(synthSnapshot(project)['babel.config.js']).toMatchSnapshot()
})

test('with filename', () => {
  const project = new Project({
    name: 'test-project',
  })

  new Babel(project, { filename: 'babel.config.cjs' })

  expect(synthSnapshot(project)['babel.config.cjs']).toMatchSnapshot()
})

test('with config API function', () => {
  const project = new Project({
    name: 'test-project',
  })

  new Babel(project, { configApiFunction: 'api.cache(true)' })

  expect(synthSnapshot(project)['babel.config.js']).toMatchSnapshot()
})

test('with blank config API function', () => {
  const project = new Project({
    name: 'test-project',
  })

  new Babel(project, { configApiFunction: '' })

  expect(synthSnapshot(project)['babel.config.js']).toMatchSnapshot()
})

test('with transform options', () => {
  const project = new Project({
    name: 'test-project',
  })

  new Babel(project, { options: { presets: ['@babel/preset-env'] } })

  expect(synthSnapshot(project)['babel.config.js']).toMatchSnapshot()
})

test('with config API function and transform options', () => {
  const project = new Project({
    name: 'test-project',
  })

  new Babel(project, {
    configApiFunction: 'api.cache(true)',
    options: { presets: ['@babel/preset-env'] },
  })

  expect(synthSnapshot(project)['babel.config.js']).toMatchSnapshot()
})
