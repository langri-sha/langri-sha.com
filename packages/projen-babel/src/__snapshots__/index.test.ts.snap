// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaults 1`] = `
"/** @type {import('@babel/core').TransformOptions} */
const config = {}

module.exports = config
"
`;

exports[`with blank config API function 1`] = `
"/** @type {import('@babel/core').ConfigFunction} */
const config = (api) => {
return {}
}

module.exports = config
"
`;

exports[`with config API function 1`] = `
"/** @type {import('@babel/core').ConfigFunction} */
const config = (api) => {
api.cache(true)

return {}
}

module.exports = config
"
`;

exports[`with config API function and transform options 1`] = `
"/** @type {import('@babel/core').ConfigFunction} */
const config = (api) => {
api.cache(true)

return {
  "presets": [
    "@babel/preset-env"
  ]
}
}

module.exports = config
"
`;

exports[`with filename 1`] = `
"/** @type {import('@babel/core').TransformOptions} */
const config = {}

module.exports = config
"
`;

exports[`with transform options 1`] = `
"/** @type {import('@babel/core').TransformOptions} */
const config = {
  "presets": [
    "@babel/preset-env"
  ]
}

module.exports = config
"
`;
